src/Features/Core/Portable/Completion/CompletionItem.cs

```diff
@@ -188,5 +189,24 @@ public override string ToString()
         {
             return DisplayText;
         }
+
+        private List<string> tags = null;
+        public IEnumerable<string> Tags
+        {
+            get
+            {
+                return tags ?? SpecializedCollections.EmptyEnumerable<string>();
+            }
+        }
+
+        public void AddTag(string tag)
```

Comment:
Completion items were immutable before.  now they're not.  that makes me sad.  Why not have 'tags' just be something that can be provided at completion item creation?